src/common/services/treeService.js

```diff
@@ -54,13 +80,22 @@ const setRulesTreeIds = (tree, rules) => {
     });
 };
 
-const findIdByObjNumbers = (node, pathArray) => {
-    if (!pathArray || !pathArray.length) return node.id;
-    const path = pathArray.pop();
-    const [num, gen] = path.split(':');
-    const nextNode = node.children.find(({ ref }) => ref.num === +num && ref.gen === +gen);
-    if (!nextNode) return null;
-    return findIdByObjNumbers(nextNode, pathArray);
+export const getNodeTitle = node => {
+    const info = {
+        objectNumber: '',
+        mcidList: '',
+    };
+
+    if (isProperty(node, 'ref')) {
+        const { num, gen } = node.ref;
+        info.objectNumber = `Object number: ${num} ${gen}`;
```

Comment:
It is not entirely clear which number is responsible for which parameter. Maybe it makes sense to call these numbers somehow? Also _gen_ is always 0.